integration_test/third_party_apps_data/applications/hadoop/centos_rhel/install

```diff
@@ -0,0 +1,144 @@
+</property>
+</configuration>
+EOF
+
+cat <<EOF | sudo tee /opt/hadoop/etc/hadoop/hadoop-env.sh
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+export JAVA_HOME=\${JAVA_HOME}
+export HADOOP_CONF_DIR=\${HADOOP_CONF_DIR:-"/etc/hadoop"}
+for f in \$HADOOP_HOME/contrib/capacity-scheduler/*.jar; do
+  if [ "\$HADOOP_CLASSPATH" ]; then
+    export HADOOP_CLASSPATH=\$HADOOP_CLASSPATH:\$f
+  else
+    export HADOOP_CLASSPATH=\$f
+  fi
+done
+export HADOOP_OPTS="\$HADOOP_OPTS -Djava.net.preferIPv4Stack=true"
+export HADOOP_NAMENODE_OPTS="-Dhadoop.security.logger=\${HADOOP_SECURITY_LOGGER:-INFO,RFAS} -Dhdfs.audit.logger=\${HDFS_AUDIT_LOGGER:-INFO,NullAppender} \$HADOOP_NAMENODE_OPTS"
+export HADOOP_NAMENODE_OPTS="-Dcom.sun.management.jmxremote \$HADOOP_NAMENODE_OPTS"
+export HADOOP_NAMENODE_OPTS="\$HADOOP_NAMENODE_OPTS -Dcom.sun.management.jmxremote.authenticate=false"
+export HADOOP_NAMENODE_OPTS="\$HADOOP_NAMENODE_OPTS -Dcom.sun.management.jmxremote.ssl=false"
+export HADOOP_NAMENODE_OPTS="\$HADOOP_NAMENODE_OPTS -Dcom.sun.management.jmxremote.port=8004 -Dcom.sun.management.jmxremote.rmi.port=8004"
+export HADOOP_DATANODE_OPTS="-Dhadoop.security.logger=ERROR,RFAS \$HADOOP_DATANODE_OPTS"
+export HADOOP_SECONDARYNAMENODE_OPTS="-Dhadoop.security.logger=\${HADOOP_SECURITY_LOGGER:-INFO,RFAS} -Dhdfs.audit.logger=\${HDFS_AUDIT_LOGGER:-INFO,NullAppender} \$HADOOP_SECONDARYNAMENODE_OPTS"
+export HADOOP_NFS3_OPTS="\$HADOOP_NFS3_OPTS"
+export HADOOP_PORTMAP_OPTS="-Xmx512m \$HADOOP_PORTMAP_OPTS"
+export HADOOP_CLIENT_OPTS="\$HADOOP_CLIENT_OPTS"
+if [ "\$HADOOP_HEAPSIZE" = "" ]; then
+  export HADOOP_CLIENT_OPTS="-Xmx512m \$HADOOP_CLIENT_OPTS"
+fi
+export HADOOP_SECURE_DN_USER=\${HADOOP_SECURE_DN_USER}
+export HADOOP_PID_DIR=\${HADOOP_PID_DIR}
+export HADOOP_SECURE_DN_PID_DIR=\${HADOOP_PID_DIR}
+export HADOOP_IDENT_STRING=\$USER
+EOF
+
+getent group hadoop || sudo groupadd hadoop
+id -u hadoop || sudo useradd -s /bin/bash -d /opt/hadoop hadoop -g hadoop
+
+sudo chown -R hadoop:hadoop /opt/hadoop
+sudo chown -R hadoop:hadoop /opt/volume
+
+sudo -u hadoop bash -c "cat /opt/hadoop/env >> ~/.bashrc"
+sudo -u hadoop bash -c "echo -e 'y\n' | ssh-keygen -b 2048 -t rsa -f ~/.ssh/id_rsa -q -N \"\""
```

Comment:
... I don't want to know why the hadoop user needs to be able to ssh to itself, do I?

Reply:
I promise it's a [requirement](https://phoenixnap.com/kb/install-hadoop-ubuntu).

Reply:
Aha, I found it: https://stackoverflow.com/a/20477333
Apparently it uses ssh to start the daemons on data nodes. I'm kinda surprised that is needed on a single-node cluster.